metabase/metabase
# FE - Make editing table metadata not rely on the full database metadata

Epic https://github.com/metabase/metabase/issues/30190

Changes:
- Uses endpoints that fetch data by schema/table instead of fetching full db metadata
- Removes special data model entities and uses `metabase-lib`
- Adds `getMetadataUnfiltered` for `getObjectUnfiltered` selectors in the entities framework to get the full graph of entities without removing hidden tables/fields etc
- Adds jest tests
- Unskips and fixes e2e tests

Markup is not changed, however data flow in the component tree is different. Here is the mapping between the new components and the old ones:

- `FieldVisibilityPicker` from `ColumnItem`
- `MetadataBackButton` from `FieldApp`
- `MetadataEditor` from `MetadataEditorApp`
- `MetadataHeader` from `MetadataHeader`
- `MetadataSchemaList` from `MetadataSchemaList`
- `MetadataSection`, `MetadataSectionHeader` from `Section`
- `MetadataTable` from `MetadataTable`
- `MetadataTableList` from `MetadataTableList`
- `MetadataTablePicker` from `MetadataTablePicker`
- `MetadataTableSchema` from `MetadataSchema`
- `MetadataTableSettings` from `TableSettingsApp`
- `MetadataTableSyncWarning` from `TableSyncWarning`
- `SemanticTypeAndTargetPicker` from `ColumnItem`

How to verify:
- Make sure updated data model pages never load all tables with their fields at once
- Hiding tables, changing fields, data model permissions (EE) should still work

Please note that the field settings page (with value remapping etc) is not yet changed.

<img width="1728" alt="Screenshot 2023-04-20 at 23 51 12" src="https://user-images.githubusercontent.com/8542534/233485042-88e349e4-6274-4add-a81e-b684c484da92.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30204)
